crates/whiskers/src/context.rs

```diff
@@ -28,10 +30,37 @@ impl Context {
 
     /// Helper function to generate a random number in a given range. This function accepts an empty
     /// range, in which case it will always return the start value.
-    pub fn rng_range(&mut self, range: Range<f64>) -> f64 {
+    pub fn rng_range<T: SampleUniform + PartialOrd>(&mut self, range: Range<T>) -> T {
         if range.is_empty() {
             return range.start;
         }
         self.rng.gen_range(range)
     }
+
+    /// Helper function to generate a random boolean value
+    pub fn rng_bool(&mut self) -> bool {
+        self.rng.gen_bool(0.5)
+    }
```

Comment:
Now I really think we should add that probability parameter as well, to be consistent with `gen_bool`.

Reply:
wouldn't that be too redundant? One could just use `ctx.rng.gen_bool`.

Reply:
Maybe you're right.

Thinking of it, I might end up adding `impl Rng for Context` (delegating to `self.rng`), so the whole thing would behave as a `rng` itself. That would entail some renaming for consistency (rng_range -> gen_range_safe? rng_bool -> gen_bool_fair?). But let's leave that for later.